tools/gengen/util/gengen.go

```diff
@@ -123,7 +123,7 @@ func GenGen(ctx context.Context, cfg *GenesisCfg, cst *hamt.CborIpldStore, bs bl
 		return nil, err
 	}
 
-	if err := setupPrealloc(st, keys, cfg.PreAlloc); err != nil {
```

Comment:
We could have avoided lots of the trouble in the consensus/genesis function if we instead used one shared genesis state / block implementation.

The function options that modify the consensus.Config could all be modified to operate over a gengen config.  The interfaces are slightly different but IMO we'd be much much better off adopting usage of the other interface to work with gengen configs OR modifying gengen configs to be slightly more expressive.

Wrote this up here: https://github.com/filecoin-project/go-filecoin/issues/3701